lib/wrapperService.ts

```diff
@@ -0,0 +1,175 @@
+  DaoCreator: DaoCreatorWrapper;
+  GenesisProtocol: GenesisProtocolWrapper;
+  GlobalConstraintRegistrar: GlobalConstraintRegistrarWrapper;
+  SchemeRegistrar: SchemeRegistrarWrapper;
+  TokenCapGC: TokenCapGCWrapper;
+  UpgradeScheme: UpgradeSchemeWrapper;
+  VestingScheme: VestingSchemeWrapper;
+  VoteInOrganizationScheme: VoteInOrganizationSchemeWrapper;
+}
+
+/**
+ * Arc.js wrapper factores grouped by type.
+ */
+export interface ArcWrappersByType {
+  /**
+   * All wrapped contracts
+   */
+  allWrappers: ArcWrappers;
+  /**
+   * All wrapped schemes
+   */
+  schemes: Array<ContractWrapperBase>;
+  /**
+   * All wrapped voting machines
+   */
+  votingMachines: Array<ContractWrapperBase>;
+  /**
+   * All wrapped global constraints
+   */
+  globalConstraints: Array<ContractWrapperBase>;
+  /**
+   * Other types of wrappers
+   */
+  other: Array<ContractWrapperBase>;
+}
+
+/**
+ * Service that provides access to Arc.js contract wrapper classes and class factories.
+ */
+export class WrapperService {
+
+  /**
+   * Wrappers by name, hydrated with contracts as deployed by this version of Arc.js.
+   */
+  public static wrappers: ArcWrappers;
+  /**
+   * Contract wrapper factories grouped by type
+   */
+  public static wrappersByType: ArcWrappersByType;
+  /**
+   * Wrapper factories by name.  Use these when you want to do `.at()` or `.new()`.  You can also
+   * use for `deployed()`, but the wrappers for deployed contracts are directly available from the
+   * `wrappers` and `wrappersByType` properties.
+   */
+  public static factories: ArcWrapperFactories = {
+    AbsoluteVote: AbsoluteVote as ContractWrapperFactory<AbsoluteVoteWrapper>,
+    ContributionReward: ContributionReward as ContractWrapperFactory<ContributionRewardWrapper>,
+    DaoCreator: DaoCreator as ContractWrapperFactory<DaoCreatorWrapper>,
+    GenesisProtocol: GenesisProtocol as ContractWrapperFactory<GenesisProtocolWrapper>,
+    GlobalConstraintRegistrar: GlobalConstraintRegistrar as ContractWrapperFactory<GlobalConstraintRegistrarWrapper>,
+    SchemeRegistrar: SchemeRegistrar as ContractWrapperFactory<SchemeRegistrarWrapper>,
+    TokenCapGC: TokenCapGC as ContractWrapperFactory<TokenCapGCWrapper>,
+    UpgradeScheme: UpgradeScheme as ContractWrapperFactory<UpgradeSchemeWrapper>,
+    VestingScheme: VestingScheme as ContractWrapperFactory<VestingSchemeWrapper>,
+    VoteInOrganizationScheme: VoteInOrganizationScheme as ContractWrapperFactory<VoteInOrganizationSchemeWrapper>,
+  };
+
+  /**
+   * initialize() must be called before any of the static properties will have values.
+   * It is currently called in ArcInitialize(), which in trun must be invoked by any applicaiton
+   * using Arc.js.
+   */
+  public static async initialize(): Promise<void> {
+    LoggingService.debug("WrapperService: initializing");
+    /**
+     * Deployed contract wrappers by name.
+     */
+    WrapperService.wrappers = {
+      AbsoluteVote: await AbsoluteVote.deployed(),
```

Comment:
why do we need to load all the deployed contracts at the beginning? seems unnecessary since most apps wont use all of them.

Reply:
I thought a lot about that.  The alternative is to load on demand.  That would mean that the WrapperService would need to be instantiated, and then something like 

```
const upgradeScheme = WrapperService.wrapper.UpgradeScheme;
```

would become:  

```
const wrapperService  = new WrapperService();
const upgradeScheme = await wrapperService.getScheme("UpgradeScheme");
```

I chose to go with the cleaner usage.

Another option could be that in `ArcInitialize()` we add a parameter that is a collection of contract names that the application intends to use, and filter what we load by that list.  Easy to do.  Similar to other libraries that allow one to customize the footprint of the package.